apps/deploy-web/src/components/new-deployment/ManifestEdit.tsx

```diff
@@ -35,12 +36,13 @@ import { Button, Alert, InputWithIcon, CustomTooltip, Spinner } from "@akashnetw
 import { SdlBuilder, SdlBuilderRefType } from "./SdlBuilder";
 
 type Props = {
+  setSelectedTemplate: Dispatch<TemplateCreation>;
```

Comment:
this name suggests that it requires a method to be passed that is defined by the `ManifestEdit` component.
However this component shouldn't know of the higher order details. This is why it's better using a convention in this case to make it an event.

```suggestion
  onTemplateSelected: Dispatch<TemplateCreation>;
```

Later on the client component would have a clearer and verbose interface to use:

```typescript
<ManifestEdit onTemplateSelected={setSelectedTemplate} />
```

I realise this is you moving same name from another place. Would be nice to improve it as we're on it.